### PR TITLE
WIP: Update msvc build to ignore warning and bump Qt version to 5.12.6.

### PR DESCRIPTION
**Pull request by sipsorcery**
WIP while checking whether the Bitcoin Core appveyor job passes. This depends on the state of the vcpkg packages which can't be replicated on a fork.

msvc warning C4834 for the Bitcoin Core build was introduced by Visual Studio 16.4.0. This PR adds an ignore rule for the warning (it's related to the nodiscard attribute and is not considered relevant).

An additional side effect of the msvc compiler update is the cached Qt5.9.8 libraries cannot be linked due to being built with an earlier version of the compiler. To fix this Qt5.12.6 has been built with the updated msvc compiler and added to the appveyor job.

### CODE REVIEW

File: build_msvc/common.init.vcxproj [hunk 1]
```diff
@@ -107,15 +107,15 @@
...
       <TreatWarningAsError>true</TreatWarningAsError>
       <PreprocessorDefinitions>ZMQ_STATIC;NOMINMAX;WIN32;HAVE_CONFIG_H;_CRT_SECURE_NO_WARNINGS;_SCL_SECURE_NO_WARNINGS;_CONSOLE;_WIN32_WINNT=0x0601;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <AdditionalIncludeDirectories>..\..\src;..\..\src\univalue\include;..\..\src\secp256k1\include;..\..\src\leveldb\include;..\..\src\leveldb\helpers\memenv;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Console</SubSystem>
       <GenerateDebugInformation>true</GenerateDebugInformation>
-      <AdditionalDependencies>crypt32.lib;Iphlpapi.lib;ws2_32.lib;Shlwapi.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>Wtsapi32.lib;crypt32.lib;Iphlpapi.lib;ws2_32.lib;Shlwapi.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;%(AdditionalDependencies)</AdditionalDependencies>
```
Comment (fanquake) [inline build_msvc/common.init.vcxproj]: > Wtsapi32.lib

Why is this needed? A quick search suggests it is the library for the `Remote Desktop Services API`.

Comment (sipsorcery) [reply build_msvc/common.init.vcxproj]: Qt5.12.6 wanted it. That did raise an orange flag for me as well. Some quick research suggested it's to allow Qt apps to be served across remote terminal connections rather than any kind of hosting ability.

**Comment by sipsorcery** [comment]
When the `Wstapi32.lib` is not referenced the linker error is:

````
"c:\Dev\github\sipsorcery_bitcoin\build_msvc\bitcoin.sln" (build target) (1) ->
"c:\Dev\github\sipsorcery_bitcoin\build_msvc\bitcoin-qt\bitcoin-qt.vcxproj" (default target) (23) ->
(Link target) ->
 qwindows.lib(qwindowscontext.obj) : error LNK2001: unresolved external symbol WTSQuerySessionInformationW [c:\
Dev\github\sipsorcery_bitcoin\build_msvc\bitcoin-qt\bitcoin-qt.vcxproj]
 qwindows.lib(qwindowscontext.obj) : error LNK2001: unresolved external symbol WTSFreeMemory [c:\Dev\github\sip
sorcery_bitcoin\build_msvc\bitcoin-qt\bitcoin-qt.vcxproj]
 c:\Dev\github\sipsorcery_bitcoin\build_msvc\x64\Release\bitcoin-qt.exe : fatal error LNK1120: 2 unresolved ext
ernals [c:\Dev\github\sipsorcery_bitcoin\build_msvc\bitcoin-qt\bitcoin-qt.vcxproj]
````

The two Remote Desktop functions being used are [WTSQuerySessionInformationW](https://docs.microsoft.com/en-us/windows/win32/api/wtsapi32/nf-wtsapi32-wtsquerysessioninformationw) and [WTSFreeMemory](https://docs.microsoft.com/en-us/windows/win32/api/wtsapi32/nf-wtsapi32-wtsfreememory).

A query and free memory call so no obvious capabilities to start a remote session or such.

I am working on a new build of Qt incorporating #17730 to see if the need for `Wstapi32.lib` can be eliminated. My gut feel is it can't since it's being called from the Qt `qwindows.lib` and based on Qt's `configure -list-features` and `configure -list-libraries` there are no obvious options related to it.

**Review by maflcko — APPROVED** [review]
ACK. Didn't look at the changes, but this makes appveyor green.

**Comment by laanwj** [comment]
> A query and free memory call so no obvious capabilities to start a remote session or such.

Ok, it's kind of worrying that Qt keeps adding these kind of requirements, but not necessary to handle here.

ACK if Appveyor passes again.

**Comment by sipsorcery** [comment]
To make this less contentious I'll rebuild Qt5.9.8 with Visual Studio 2019 v16.4 and resubmit the PR without needing to bump Qt.
